diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -290,7 +290,7 @@ endif()
 # ---[ EIGEN
 # Due to license considerations, we will only use the MPL2 parts of Eigen.
 set(EIGEN_MPL2_ONLY 1)
-find_package(Eigen3)
+find_package(Eigen3 3.3.0)
 if(EIGEN3_FOUND)
   message(STATUS "Found system Eigen at " ${EIGEN3_INCLUDE_DIR})
 else()
@@ -381,6 +381,7 @@ if(USE_CUDA)
     # A helper variable recording the list of Caffe2 dependent libraries
     # caffe2::cudart is dealt with separately, due to CUDA_ADD_LIBRARY
     # design reason (it adds CUDA_LIBRARIES itself).
+    include_directories(${CUDNN_INCLUDE_DIR})
     set(Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS
         caffe2::cuda caffe2::curand caffe2::cudnn caffe2::nvrtc)
     if(CAFFE2_STATIC_LINK_CUDA)
